drivers/pcie/host/pcie.c

```diff
@@ -291,8 +298,16 @@ unsigned int pcie_alloc_irq(pcie_bdf_t bdf)
 	    irq >= CONFIG_MAX_IRQ_LINES ||
 	    arch_irq_is_used(irq)) {
 
+	/* In some platforms, PCI interrupts are hardwired to specific interrupt inputs
+	 * on the interrupt controller and are not configurable. Hence we need to retrieve
+	 * IRQ from acpi. But if it is configurable then we allocate irq dynamically.
+	 */
 #ifdef CONFIG_PCIE_PRT
-		irq = acpi_legacy_irq_get(bdf);
+		if (prt_base_irq) {
+			irq = acpi_legacy_irq_get(bdf);
+		} else {
+			irq = arch_irq_allocate();
+		}
 #else
 		irq = arch_irq_allocate();
 #endif
```

Comment:
rewrite the above so it becomes:
```
#ifdef CONFIG_PCIE_PRT
		irq = acpi_legacy_irq_get(bdf);
		if (prt_base_irq) {
			irq = acpi_legacy_irq_get(bdf);
		} else
#else
                {
			irq = arch_irq_allocate();
		}
#endif 
```

Reply:
> rewrite the above so it becomes:

@tbursztyka I don't think what you proposed will generate correct code, since it leaves out the `arch_irq_allocate()` call completely if `CONFIG_PCIE_PRT` is set. Anyway, think Najumon's original code is actually a bit more readable, even though there`s a `arch_irq_allocate()` in two places in the code.

Reply:
> > rewrite the above so it becomes:
> 
> @tbursztyka I don't think what you proposed will generate correct code, since it leaves out the `arch_irq_allocate()` call completely if `CONFIG_PCIE_PRT` is set. Anyway, think Najumon's original code is actually a bit more readable, even though there's `arch_irq_allocate()` in two places.

updated with slight optimization